server/routes/install-routes.js

```diff
@@ -117,22 +118,27 @@ function installRoutes(app) {
       // the format isn't correct, redirect to the correct hour range
       if (hourRange && toHour !== fromHour + 1) {
         res.redirect(
-          urlJoin(
+          // FIXME: Can we use the matrixPublicArchiveURLCreator here?
+          `${urlJoin(
             basePath,
             roomIdOrAlias,
             'date',
             req.params.yyyy,
             req.params.mm,
             req.params.dd,
             `${fromHour}-${fromHour + 1}`
-          )
+          )}?${new URLSearchParams(req.query).toString()}`
         );
         return;
       }
 
       // TODO: Highlight tile that matches ?at=$xxx
       //const aroundId = req.query.at;
 
+      // We have to wait for the room join to happen first before we can fetch
+      // any of the additional room info or messages.
+      await ensureRoomJoined(matrixAccessToken, roomIdOrAlias, req.query.via);
```

Comment:
This takes some time even when you're already joined to the room. Is there a faster way to determine whether we are already joined?

Maybe we can just go for it and if we get a `403`/`401`, we can join and re-run the data fetchers.

Reply:
Tracked by https://github.com/matrix-org/matrix-public-archive/issues/50